include/gridtools/common/integral_constant.hpp

```diff
@@ -41,6 +41,19 @@ namespace gridtools {
         constexpr GT_FUNCTION operator T() const noexcept { return V; }
     };
 
+    // Returns the value_type of an integral_constant-like or returns the type itself
+    template <class T, class U = void>
+    struct to_integral_type {
+        using type = T;
+    };
```

Comment:
Probably it makes sense to enable this only if `std::is_integral_v` is true?

Reply:
done